### PR TITLE
Remove rocblas on windows

### PR DESCRIPTION
**Pull request by tvukovic-amd**
Added MIGRAPHX_USE_ROCBLAS build option which is automatically ON on Windows and OFF on Linux.
When MIGRAPHX_USE_ROCBLAS is OFF, rocBLAS is removed on MIGraphX.

@apwojcik

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -340,5 +346,10 @@ rocm_create_package(
     MAINTAINER "AMDMIGraphX Maintainer <migraphx-lib.support@amd.com>"
     LDCONFIG
     PTH
-    DEPENDS miopen-hip rocblas ${DEPENDS_HIP_RUNTIME} hip-base half ${PACKAGE_DEPENDS}
+    DEPENDS miopen-hip ${DEPENDS_HIP_RUNTIME} hip-base half ${PACKAGE_DEPENDS}
 )
+
+if(MIGRAPHX_USE_ROCBLAS)
+    list(APPEND PACKAGE_DEPENDS PACKAGE rocblas)
+endif()
```
Comment (pfultz2) [inline CMakeLists.txt:354]: This needs to be done before `rocm_create_package`.

File: src/targets/gpu/compile_miopen.cpp [hunk 2]
```diff
@@ -29,7 +29,9 @@
 #include <migraphx/make_op.hpp>
 #include <migraphx/register_op.hpp>
 #include <migraphx/op/identity.hpp>
-#include <migraphx/gpu/rocblas.hpp>
+#ifdef MIGRAPHX_USE_ROCBLAS
```
Comment (pfultz2) [inline src/targets/gpu/compile_miopen.cpp:32]: This should be defined as 1 or 0, so it should use `#if MIGRAPHX_USE_ROCBLAS`. Also I dont see you adding this define to cmake.

File: src/targets/gpu/fuse_mlir.cpp [hunk 3]
```diff
@@ -79,11 +81,23 @@ static bool is_negated_op(const std::string& s)
     return contains({'!', '~'}, s[0]);
 }
 
+#ifndef MIGRAPHX_USE_ROCBLAS
+const std::vector<std::string>& get_default_options()
+{
+    static std::vector<std::string> default_options{"dot", "convolution", "fused"};
+    return default_options;
+}
+#endif
```
Comment (pfultz2) [inline src/targets/gpu/fuse_mlir.cpp:90]: We should not be changing how the environment variable works. This also disables miopen convolutions as well, which seems unrelated.

File: src/targets/gpu/fuse_ops.cpp [hunk 4]
```diff
@@ -893,11 +898,13 @@ void fuse_ops::apply(module& m) const
     match::find_matches(m, find_conv_pointwise{ctx}, find_conv_bias_relu{ctx}, find_conv_bias{ctx});
     run_passes(m, {dead_code_elimination{}});
     match::find_matches(m,
-                        find_gemm_pointwise{},
                         find_layernorm_pointwise{},
                         find_concat_pointwise{},
                         find_contiguous_tranpose_gemm{},
                         find_commutative_broadcast{});
+    #ifdef MIGRAPHX_USE_ROCBLAS
+        match::find_matches(m, find_gemm_pointwise{});
```
Comment (pfultz2) [inline src/targets/gpu/fuse_ops.cpp:906]: This should not be moved here, it should stay in the list where it was at if rocblas is enabled.

File: src/targets/gpu/include/migraphx/gpu/context.hpp [hunk 5]
```diff
@@ -27,7 +27,9 @@
 #include <migraphx/gpu/export.h>
 #include <migraphx/context.hpp>
 #include <migraphx/gpu/miopen.hpp>
+#if MIGRAPHX_USE_ROCBLAS
 #include <migraphx/gpu/rocblas.hpp>
+#endif
```
Comment (pfultz2) [inline src/targets/gpu/include/migraphx/gpu/context.hpp:32]: You already ifdef this file so you shouldn't need an ifdef when including it.

File: src/targets/gpu/lowering.cpp [hunk 6]
```diff
@@ -47,7 +47,9 @@
 #include <migraphx/gpu/device_name.hpp>
 #include <migraphx/gpu/gemm.hpp>
 #include <migraphx/gpu/miopen.hpp>
+#if MIGRAPHX_USE_ROCBLAS
 #include <migraphx/gpu/rocblas.hpp>
+#endif
```
Comment (pfultz2) [inline src/targets/gpu/lowering.cpp:52]: Same here, this doesnt need a ifdef.

File: src/targets/gpu/compile_miopen.cpp [hunk 7]
```diff
@@ -29,7 +29,9 @@
 #include <migraphx/make_op.hpp>
 #include <migraphx/register_op.hpp>
 #include <migraphx/op/identity.hpp>
+#if MIGRAPHX_USE_ROCBLAS
 #include <migraphx/gpu/rocblas.hpp>
+#endif
```
Comment (pfultz2) [inline src/targets/gpu/compile_miopen.cpp:34]: Same here, the ifdef can be removed.

File: src/targets/gpu/CMakeLists.txt [hunk 8]
```diff
@@ -112,6 +116,10 @@ if(NOT MIGRAPHX_USE_COMPOSABLEKERNEL)
             ${CMAKE_CURRENT_SOURCE_DIR}/jit/ck_gemm_softmax_gemm.cpp)
 endif()
 
+if (MIGRAPHX_USE_ROCBLAS)
+    set(ROCBLAS_SRCS rocblas.cpp gemm_impl.cpp)
```
Comment (pfultz2) [inline src/targets/gpu/CMakeLists.txt:120]: You already ifdef these sources, so we dont need to remove them from the list in cmake.

**Comment by tperry-amd** [comment]
Please change the code to default the CMake variable to ON for now.

File: CMakeLists.txt [hunk 9]
```diff
@@ -61,6 +61,12 @@ else()
 option(MIGRAPHX_ENABLE_PYTHON "Enable python bindings" ON)
 endif()
 
+if(WIN32)
```
Comment (tperry-amd) [inline CMakeLists.txt]: Let's change this code to default the variable to ON only right now.

**Review by tperry-amd — CHANGES_REQUESTED** [review]

**Review by tperry-amd — APPROVED** [review]

**Pull request merged**